src/beta/realtime/websocket.ts

```diff
@@ -95,3 +95,106 @@ export class OpenAIRealtimeWebSocket extends OpenAIRealtimeEmitter {
     }
   }
 }
+
+export class AzureOpenAIRealtimeWebSocket extends OpenAIRealtimeEmitter {
+  socket: _WebSocket;
+
+  constructor(
+    private client: AzureOpenAI,
+    private options: {
+      deploymentName?: string;
+    } = {},
+  ) {
+    super();
+  }
+
+  async open(): Promise<void> {
+    async function getUrl({
```

Comment:
nit: can you move this function outside of this class? would be much easier to read